alphastats/gui/utils/analysis_helper.py

```diff
@@ -79,26 +85,23 @@ def download_preprocessing_info(plot):
     )
 
 
-def read_uploaded_file_into_df(file, decimal="."):
-    filename = file.name
+def _read_file_to_df(file: UploadedFile, decimal: str = ".") -> Optional[pd.DataFrame]:
+    """Read file to DataFrame based on file extension."""
 
-    if filename.endswith(".xlsx"):
-        df = pd.read_excel(file)
+    extension = Path(file.name).suffix
 
-    elif filename.endswith(".txt") or filename.endswith(".tsv"):
-        df = pd.read_csv(file, delimiter="\t", decimal=decimal)
+    if extension == ".xlsx":
+        return pd.read_excel(file)
 
-    elif filename.endswith(".csv"):
-        df = pd.read_csv(file, decimal=decimal)
+    elif extension in [".txt", ".tsv"]:
+        return pd.read_csv(file, delimiter="\t", decimal=decimal)
 
-    else:
-        df = None
-        logging.warning(
-            "WARNING: File could not be read. \nFile has to be a .xslx, .tsv, .csv or .txt file"
-        )
-        return
+    elif extension == ".csv":
+        return pd.read_csv(file, decimal=decimal)
 
-    return df
+    raise ValueError(
+        f"Unknown file type '{extension}'. \nSupported types: .xslx, .tsv, .csv or .txt file"
```

Comment:
How does this appear in the logging output in the end? Do we even need the logging library?

Reply:
the error message should be logged automatically .. the logging lib still has value to add some `INFO` statements that might help